src/libcollections/string.rs

```diff
@@ -386,12 +418,14 @@ impl String {
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
     /// // 𝄞music
     /// let v = &[0xD834, 0xDD1E, 0x006d, 0x0075,
     ///           0x0073, 0x0069, 0x0063];
-    /// assert_eq!(String::from_utf16(v).unwrap(),
-    ///            "𝄞music".to_string());
+    /// assert_eq!(Some(String::from("𝄞music")),
+    ///            String::from_utf16(v));
```

Comment:
> I also think that String::from may be a bit distracting here when compared to to_string?

I happen to like it _more_ in this case, as it's closely named with `from_utf16`.

@tbu- this isn't generic code, so `.to_owned` isn't appropriate.